### PR TITLE
Snowball update

### PR DESCRIPTION
**Pull request by GrateBrittan**
Should make the snowball and its effects the color that has been selected by the player (Now we can have yellow snow ;))

### CODE REVIEW

File: lua/entities/ent_snowball_nodamage.lua [hunk 1]
```diff
@@ -41,19 +41,22 @@ function ENT:Draw()
...
 		data.HitEntity:SetVelocity(Vector(fwd * 100))
 	end
 
-	local p1 = data.HitPos + (data.HitNormal * 2)
-	local p2 = data.HitPos - (data.HitNormal * 2)
-	util.Decal("Splash.Large", p1, p2)
+	--local p1 = data.HitPos + (data.HitNormal * 2)
+	--local p2 = data.HitPos - (data.HitNormal * 2)
+	--util.Decal("Splash.Large", p1, p2)
+	util.DecalEx(Material("Splash.Large"), nil, data.HitPos, data.HitNormal, plycol, 1, 1)
```
Comment (swampservers) [inline lua/entities/ent_snowball_nodamage.lua]: Should be `util.DecalMaterial` not `Material`, and also, cache it outside the function

File: lua/weapons/weapon_snowball.lua [hunk 2]
```diff
@@ -68,6 +68,7 @@ function SWEP:PrimaryAttack()
 			local front = self.Owner:GetAimVector()
 			ball:SetOwner(self.Owner)
 			ball:SetPos(self.Owner:GetShootPos() + front * 10 + self.Owner:EyeAngles():Up() * -5)
+			Ball:SetColor(self.Owner:GetNWVector("SnowballColor"), Vector(1, 1, 1)):ToColor)
```
Comment (swampservers) [inline lua/weapons/weapon_snowball.lua]: you're missing parens after :ToColor and you have an extra close paren after "SnowballColor"

Comment (GrateBrittan) [reply lua/weapons/weapon_snowball.lua]: Okay this should now be fixed

File: lua/entities/ent_snowball_nodamage.lua (hunk 1 above)
Comment (GrateBrittan) [reply lua/entities/ent_snowball_nodamage.lua]: okay made the changes, hopefully I did them right

**Pull request merged**